recommendation_engine/pmf/train_pmf.py

```diff
@@ -0,0 +1,157 @@
+                                                 transpose_a=True) * training_params.b
+
+                for item_id, this_item_users in enumerate(item_user_map):
+                    if len(this_item_users) == 0:
+                        # Never been rated
+                        item_norm = items_users_weighted + tf.eye(
+                                training_params.num_latent) * training_params.lambda_v
+                        self.m_items[item_id, :] = tf.linalg.solve(
+                                item_norm,
+                                training_params.lambda_v * self.m_weights[item_id, :])
+
+                        # now calculate the likelihood
+                        epsilon = self.m_items[item_id, :] - self.m_weights[item_id, :]
+                        likelihood += -0.5 * training_params.lambda_v * tf.reduce_sum(
+                                tf.square(epsilon))
+                    else:
+                        item_norm = items_users_weighted + tf.matmul(
+                                tf.gather(self.m_users, this_item_users),
+                                tf.gather(self.m_users, this_item_users),
+                                transpose_a=True) * a_minus_b
+
+                        item_norm_pre_weighing = item_norm
+                        item_norm += tf.eye(training_params.num_latent) * training_params.lambda_v
+
+                        tf.assign(self.m_items[item_id:], tf.linalg.solve(
+                                item_norm,
+                                training_params.a * tf.reduce_sum(
+                                        tf.gather(self.m_users, this_item_users),
+                                        axis=0) + training_params.lambda_v * self.m_weights[item_id,
+                                                                             :]))
+
+                        likelihood += (-0.5) * len(user_ids) * training_params.a + \
+                                      training_params.a * tf.reduce_sum(
+                                tf.matmul(tf.gather(self.m_users, user_ids),
+                                          tf.reshape(self.m_items[item_id, :],
+                                                     [training_params.num_latent, 1])), axis=0)
+
+                        likelihood += -0.5 * tf.matmul(
+                                tf.matmul(self.m_items[item_id, :], item_norm_pre_weighing),
+                                tf.reshape(self.m_items[item_id, :],
+                                           [training_params.num_latent, 1]))
+
+                        epsilon = self.m_items[item_id, :] - self.m_weights[item_id, :]
+                        likelihood += (-0.5) * training_params.lambda_v * tf.reduce_sum(
+                                tf.square(epsilon))
+
+                iteration += 1
+                convergence = tf.assign(convergence,
+                                        tf.abs((likelihood - likelihood_old) / likelihood_old))
```

Comment:
Here, likelihood_old is 0. So convergence is getting infinite. So how should we update the variable?